sky/packages/sky/lib/widgets/scaffold.dart

```diff
@@ -152,12 +153,31 @@ class RenderScaffold extends RenderBox {
     }
   }
 
+  static Point _transformPoint(Matrix4 transform, Point point) {
+    Vector3 position3 = new Vector3(point.x, point.y, 0.0);
+    Vector3 transformed3 = transform.transform3(position3);
+    return new Point(transformed3.x, transformed3.y);
+  }
+
+  Point parentToLocal(RenderBox box, Point point) {
+    assert(attached);
+    Matrix4 transform = new Matrix4.identity();
+    box.applyPaintTransform(transform);
+    /* double det = */ transform.invert();
+    // TODO(abarth): Check the determinant for degeneracy.
+    return _transformPoint(transform, point);
+  }
+
   void hitTestChildren(HitTestResult result, { Point position }) {
     for (ScaffoldSlots slot in ScaffoldSlots.values.reversed) {
       RenderBox box = _slots[slot];
       if (box != null) {
         assert(box.parentData is BoxParentData);
-        if ((box.parentData.position & box.size).contains(position)) {
+        // Apply the box's transform to check if it contains position.
+        // But when we pass the position to box.hitTest, we only want to apply
+        // the top-level transform (box will apply its own transforms).
+        Point local = parentToLocal(box, position);
+        if ((Point.origin & box.size).contains(local)) {
```

Comment:
It's ok to land this as-is, but this isn't the right solution to this problem in general.  We probably need to change how hit testing works a bit to solve this problem in all the places it occurs.